### PR TITLE
[KernelGen][Nvidia] Add lu_unpack operator with Triton kernel

### PR DESCRIPTION
**Pull request by LoserCheems**
## Description
Add `lu_unpack` and `lu_unpack_out` operators with Triton kernel implementation. This operator unpacks the LU decomposition returned by `lu_factor` into separate permutation matrix P, lower triangular L, and upper triangular U.

## Implementation Details
- **P kernel (hybrid strategy)**: Small matrices (m≤512) use a fully vectorized kernel that processes the entire permutation array in one program with SIMD-width swap operations; large matrices (m>512) use per-row scalar kernel for better parallelism.
- **L kernel**: Row-parallel extraction with vectorized load/store (BLOCK_K width).
- **U kernel**: Row-parallel extraction with vectorized load/store (BLOCK_N width).
- Supports batched inputs with arbitrary leading batch dimensions.
- Handles `unpack_data` and `unpack_pivots` flags per PyTorch API.
- Includes `lu_unpack.out` variant for pre-allocated output tensors.

### float32
| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| [64, 64] | 0.015 | 0.009 | 1.637x |
| [256, 256] | 0.032 | 0.017 | 1.869x |
| [1024, 1024] | 0.109 | 0.123 | 0.884x |
| [4096, 4096] | 0.612 | 1.706 | 0.359x |
| [1024, 65536] | 0.771 | 0.724 | 1.065x |

### float64
| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| [64, 64] | 0.014 | 0.010 | 1.442x |
| [256, 256] | 0.032 | 0.018 | 1.805x |
| [1024, 1024] | 0.122 | 0.131 | 0.928x |
| [4096, 4096] | 0.689 | 1.804 | 0.382x |
| [1024, 65536] | 0.878 | 0.975 | 0.901x |

**Mean speedup (float32): 1.16x | Mean speedup (float64): 1.09x** (5 cases each, --level core)

**Note:** Small/medium matrices (64, 256) achieve 1.4-1.9x speedup. The 4096×4096 case involves O(m×k) serial pivot operations in the per-row kernel which is compute-bound. For rectangular matrices (1024×65536), performance is on par with PyTorch.

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS (110/110) | PASS (10 cases, --level core) | 1.13x | float32 + float64 |
| Tianshu | — | — | — | Not tested |
| Muxi | — | — | — | Not tested |
| Ascend | — | — | — | Not tested |
| Hygon | — | — | — | Not tested |

Tested-on: Nvidia

### CODE REVIEW

File: src/flag_gems/ops/lu_unpack.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+
+        # Hybrid strategy based on matrix size:
+        # - Small m (<=512): per-row kernel with single store (better SM utilization)
+        # - Large m (>512): vectorized kernel processing entire perm in one program
+        #   (reduces total pivot loop iterations from m*k to k with SIMD width m)
+        if m <= 512:
+            # Per-row kernel: each program tracks one row through pivots
+            BLOCK_SIZE = 1  # Each program writes single element
+            grid = (batch_size * m,)
+            lu_unpack_p_kernel_large[grid](
```
Comment (Caeruleann) [inline src/flag_gems/ops/lu_unpack.py]: m <= 512 run lu_unpack_p_kernel_large?
m > 512 run lu_unpack_p_kernel_small ?
Did you write it wrong?

File: tests/test_lu_unpack.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import pytest
+import torch
+
+import flag_gems
+
+from . import accuracy_utils as utils
+
+# lu_factor only supports float32 in CUDA environment
+LU_SHAPES = [(3, 3), (4, 4), (3, 4), (4, 3), (5, 5), (8, 8)]
```
Comment (Caeruleann) [inline tests/test_lu_unpack.py]: Test cases do not cover paths m >512.

**Comment by Caeruleann** [comment]
please add lu_unpack_out.

File: benchmark/test_lu_unpack.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        A = torch.randn(m, n, dtype=dtype, device=device)
+        LU, pivots = torch.linalg.lu_factor(A)
+        yield LU, pivots
+
+    bench = base.GenericBenchmark2DOnly(
+        input_fn=lu_unpack_input_fn,
+        op_name="lu_unpack",
+        torch_op=torch.ops.aten.lu_unpack,
+        # torch.linalg.lu_factor only supports float32/float64; half/bfloat16 not supported
+        dtypes=[torch.float32],
```
Comment (Caeruleann) [inline benchmark/test_lu_unpack.py]: Why not test the fp64 type?

**Review by Caeruleann — APPROVED** [review]
